wmidump.c

```diff
@@ -181,6 +181,8 @@ static char *parse_ascii_wdg(const char *wdg, size_t *bytes)
 		if (errno == ERANGE && (lval == LONG_MAX || lval == LONG_MIN))
 			errx(1, "<stdin>:%ld:%ld: invalid hex number",
 			    lno, cno);
+		if (*end == '\0')
+			end--;
 		wdg += end - wdg;
```

Comment:
What is the failure scenario here? Under normal circumstances, when end is not NULL and the whole string is valid, *end will be \0. So this appears to be shortening the length of the valid string by 1 when it shouldn't. e.g.

wdg = "123"

['1']['2']['3']['\0']

wdg points to '1', end would point to '\0', but you move end to point to '3' instead, so the wdg increment becomes 2 instead of 3 as I'd expect. I presume I am missing something, but I'm not sure what it is.